File: provider-oracle/db/models/supportedpairs.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class SupportedPairs extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // define association here
    }
  };
  SupportedPairs.init({
    name: DataTypes.STRING,
    base: DataTypes.STRING,
    target: DataTypes.STRING
  }, {
    sequelize,
    modelName: 'SupportedPairs',
  });
  return SupportedPairs;
};

File: go-ooo/database/models/supported_pairs.go
package models

import "gorm.io/gorm"

type SupportedPairs struct {
	gorm.Model
	Name   string `gorm:"index"`
	Base   string `gorm:"index"`
	Target string `gorm:"index"`
}

func (SupportedPairs) TableName() string {
	return "supported_pairs"
}

func (d SupportedPairs) GetId() uint {
	return d.ID
}

func (d SupportedPairs) GetName() string {
	return d.Name
}

func (d SupportedPairs) GetBase() string {
	return d.Base
}

func (d SupportedPairs) GetTarget() string {
	return d.Target
}
